manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/PodSpecEnvHandler.java

```diff
@@ -67,7 +66,8 @@
             }
 
             // Preparing Pod Spec secrets from props
-            Secrets secrets = SecretsProvider.getSecrets(serviceSpec);
+         //   Secrets secrets = SecretsProvider.getSecrets(serviceSpec);
+            Secrets secrets = serviceSpec.get(HyscaleSpecFields.secrets, Secrets.class);
```

Comment:
Secrets have to be retrieved from Provider because it can have either a list / key-value pairs and we cannot deserialise directly to list/map . Can you cross-check once about this change ?